ui/src/main/java/org/pentaho/di/ui/spoon/delegates/SpoonSlaveDelegate.java

```diff
@@ -70,80 +80,194 @@ public void addSpoonSlave( SlaveServer slaveServer ) {
     tabfolder.setSelected( idx );
   }
 
-  public void delSlaveServer( HasSlaveServersInterface hasSlaveServersInterface, SlaveServer slaveServer )
-    throws KettleException {
+  public void delSlaveServer( SlaveServerManagementInterface slaveServerManager, SlaveServer slaveServer  ) throws KettleException {
 
-    Repository rep = spoon.getRepository();
-    if ( rep != null && slaveServer.getObjectId() != null ) {
-      // remove the slave server from the repository too...
-      rep.deleteSlave( slaveServer.getObjectId() );
-      if ( sharedObjectSyncUtil != null ) {
-        sharedObjectSyncUtil.deleteSlaveServer( slaveServer );
-      }
+    MessageBox mb = new MessageBox( spoon.getShell(), SWT.YES | SWT.NO | SWT.ICON_QUESTION );
+    mb.setMessage( BaseMessages.getString( PKG, "Spoon.Message.DeleteSlaveServerAsk.Message", slaveServer.getName() ) );
+    mb.setText( BaseMessages.getString( PKG, "Spoon.ExploreDB.DeleteConnectionAsk.Title" ) );
+    int response = mb.open();
+
+    if ( response != SWT.YES ) {
+      return;
     }
-    hasSlaveServersInterface.getSlaveServers().remove( slaveServer );
-    refreshTree();
+    spoon.getLog().logBasic( "Deleting the slave server " +  slaveServer.getName() );
+
+    try {
+      slaveServerManager.remove( slaveServer );
 
+      Repository rep = spoon.getRepository();
+      if ( rep != null && slaveServer.getObjectId() != null ) {
+        // remove the slave server from the repository too...
```

Comment:
This is going to be interesting. See the recent spike findings. Nothing to worry about in this PR, though.